pkg/prometheus/promcfg.go

```diff
@@ -2290,6 +2291,15 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	relabelings := initRelabelings()
 	labeler := namespacelabeler.New(cpf.EnforcedNamespaceLabel, cpf.ExcludedFromEnforcement, false)
 
+	if sc.Spec.JobName != "" {
+		relabelings = append(relabelings, yaml.MapSlice{
+			{Key: "target_label", Value: "scrapeJob"},
```

Comment:
```suggestion
			{Key: "target_label", Value: "job"},
```

Reply:
Could you explain why? I just tried it and now the selector from linkerd-viz can't find the metrics → they are expected at {scrapeJob="linkerd-*"} which isn't fulfilled anymore after this change...

Reply:
The `job_name` field from the Prometheus configuration translates to the `job` label. I'm not sure why linkerd-viz would use `scrapeJob` instead.

Reply:
hm, what about the label `scrapeJob=scrapeConfig/<namespace>/<metadata.name>`?

If config `job_name` translates to label `job`, would we need another relabel_config here?

Reply:
Again I'm not sure where this `scrapeJob` label comes from but it's nothing conventional. If you really need, you can still inject it with another relabel config (e.g. copy from the `job` label).

Reply:
oh, sorry, I thought it was the default for `scrapeJob`.

Well, if not I will have to stick with the relabelling on linkerd-viz. I guess I will debug on their side as well why they use it.

I fixed it with #7a2b39b5